iroha-cli/interactive/impl/interactive_query_cli.cpp

```diff
@@ -168,59 +173,67 @@ namespace iroha_cli {
           local_time_, creator_, counter_, account_id, asset_id);
     }
 
-    nonstd::optional<iroha::model::Pager> InteractiveQueryCli::parsePager(
+    iroha::model::Pager InteractiveQueryCli::parsePager(
         const std::string &encoded_tx_hash, const std::string &limit_str) {
-      return nonstd::make_optional<iroha::model::Pager>() |
-             [&encoded_tx_hash](iroha::model::Pager pager)
-               -> nonstd::optional<iroha::model::Pager> {
-               /**
-                * Use '-' as the argument of tx_hash to specify that
-                * the hash is empty because interactive cli splits
-                * arguments by space.
-                */
-               const auto PAGER_TX_HASH_EMPTY = "-";
-               if (encoded_tx_hash == PAGER_TX_HASH_EMPTY) {
-                 pager.tx_hash.fill(0);
-               } else {
-                 const auto decoded_hash =
-                   iroha::hexstringToBytestring(encoded_tx_hash);
-                 if (not decoded_hash) {
-                   return nonstd::nullopt;
-                 }
-                 pager.tx_hash = iroha::hash256_t::from_string(*decoded_hash);
-               }
-               return pager;
-             } | [&limit_str](auto pager) -> nonstd::optional<iroha::model::Pager> {
-        try {
-          pager.limit = boost::lexical_cast<decltype(pager.limit)>(limit_str);
-        } catch (...) {
-          return nonstd::nullopt;
+      iroha::model::Pager pager;
+      if (encoded_tx_hash.empty()) {
+        pager.tx_hash.fill(0);
+      } else {
+        const auto decoded_hash = iroha::hexstringToBytestring(encoded_tx_hash);
+        if (decoded_hash) {
+          pager.tx_hash = iroha::hash256_t::from_string(*decoded_hash);
+        } else {
+          log_->warn(
+              "Specified hash size is invalid. Set empty hash as "
+              "default");
+          pager.tx_hash.fill(0);
         }
-        return pager;
-      };
+      }
+      try {
+        pager.limit = boost::lexical_cast<decltype(pager.limit)>(limit_str);
+      } catch (...) {
+        log_->warn(
+            "Specified tx limit is invalid. Set {} (max size) as default",
+            MAX_PAGER_LIMIT);
+        pager.limit = MAX_PAGER_LIMIT;
```

Comment:
Same as before.

Reply:
Answered in https://github.com/hyperledger/iroha/pull/830#discussion_r173097242